Feature: add donation confirmation settings UI

jonwaldstein opened the pull request:
<!-- Make sure to prefix the title with one of New:, Fix:, Changed:, or Security: -->

<!-- Indicate the issue(s) resolved by this PR. -->

## Description

<!-- Summarize the related issue, explain HOW this PR solves the problem, and WHY you made the choices you made. -->

This updates the donation confirmation settings UI with a popover and list of available template tags (pulled from v2 forms).

## Affects

<!-- Mention any existing functionality affected by this PR to help inform the reviewer(s). -->

The donation confirmation settings UI in the form builder.  Also the confirmation page now uses Interweave to display the rich text from the template tags that were generated.

## Visuals

<!-- Include screenshots or video to better communicate your changes. -->

<img width="743" alt="Screenshot 2023-08-03 at 12 40 15 PM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/5f6f3551-38e8-4b7d-9a02-6bc0d776e4e3">

## Testing Instructions

<!-- Help others test your PR as efficiently as possible. -->

- Build assets and view a donation form builder
- Interact with the donation confirmation settings
- Make sure v2 form template tags work as expected

## Pre-review Checklist

<!-- Complete tasks prior to requesting a review. Add to this list, but do not remove the base items. -->

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203889335107900

jonwaldstein commented:
@glaubersilva this ready for review.

glaubersilva reviewed (changes requested):
@jonwaldstein Great work! About the code, I left just one comment about a minor change. 

But about the feature itself, we have a few things to take care of:

1. The line breaks don't work on the confirmation page
2. The P2P tags are loaded, but next gen will be released without compatibility with the v3 forms, more details here: https://github.com/impress-org/give-peer-to-peer/pull/505
3. The PDF receipt link is using the same color as the header background and because of that seems to not be present at the first look. 

Check the attached screenshot for visual reference:

![tag-problems](https://github.com/impress-org/givewp-next-gen/assets/16308864/174b347d-104b-442e-962b-12a25e81e9ea)

jonwaldstein commented:
@glaubersilva thanks for brining these up.  I discovered a few things:

1. _The line breaks don't work on the confirmation page_: they actually do! but some tags like pdf_receipt is making it not work.  It might be a setting on Interweave or the way we handle converting tags with links. I'll look into that.
2. _The P2P tags are loaded, but next gen will be released without compatibility with the v3 form..._ we can try to filter those out - i'll look into that.
3. Ah yeah, we can add a little styling help for links on the classic design

jonwaldstein commented on src/FormBuilder/resources/js/form-builder/src/settings/donation-confirmation/index.tsx:
```diff
@@ -0,0 +1,73 @@
+    } = usePopoverState();
+
+    return (
+        <PanelBody title={__('Donation Confirmation')} initialOpen={false}>
```
resolved 4bab9e37

jonwaldstein commented:
@glaubersilva okay man I fixed those main issues.

- Added nl2br for the content so interweave can handle line breaks appropriately alongside anchor tags
- Filtered available tag list
- Styling for me was okay on the link, let's add a QA item to make the confirmation page smarter for contrast in background color and text.  Right now it's going to depend on the design options.

<img width="794" alt="Screenshot 2023-08-09 at 4 57 04 PM" src="https://github.com/impress-org/givewp-next-gen/assets/10138447/4af0a68c-b6b4-4233-bfca-8411c0f25015">

glaubersilva commented:
@jonwaldstein Thanks for the changes, now the line breaks are working, but the link keeps the same background color on my end. About the P2P tags, a weird bug is happening when the addon is activated, check the video to see what's happening: https://www.loom.com/share/ec528d56ae124311809fcd6067b80ffe

jonwaldstein commented:
> @jonwaldstein Thanks for the changes, now the line breaks are working, but the link keeps the same background color on my end. About the P2P tags, a weird bug is happening when the addon is activated, check the video to see what's happening: https://www.loom.com/share/ec528d56ae124311809fcd6067b80ffe

@glaubersilva I just pushed a fix for the p2p conflict.  please try again.
